packages/devtools_app/lib/src/shared/table/table.dart

```diff
@@ -82,8 +82,11 @@ class FlatTable<T> extends StatefulWidget {
 
   /// List of columns to render.
   ///
-  /// If different column instances are provided for two instances of the table
-  /// with the same [dataKey], the table will render first instances.
+  /// If different column instances are provided to different instances of the table
+  /// all table instances will render first instance of the provided column type.
+  ///
+  /// So, if you have many instances of a concrete table, it is recommended to
+  /// make columns static and share between table instances.
```

Comment:
Please update this comment to the following. 

These [ColumnData] elements should be defined as static OR if they cannot be defined as static, they should not manage stateful data. [FlatTableState.didUpdateWidget] checks if the columns have changed before re-initializing the table controller, and the columns are compared by title only.

See also [FlatTableState. _tableConfigurationChanged].